### PR TITLE
Add searching by event in Editor Settings shortcuts and Project Settings input map.

### PR DESCRIPTION
**Pull request by EricEzaM**
* Focus into the `LineEdit`, then perform input to search the list of events by the events assigned.
* New specialised editor-only control for this: `EventListenerLineEdit`. Line edit is a good candidate for such a control because you can focus it, override it's input handling, and show the event all in one control.

Update `InputEventConfigurationDialog` to use event listener line edit rather than the separate tabs.
* Cleaner look - no need for tabs.
* Simpler code.

Please note the preview text in the LineEdit has now been changed from what is in the videos below. New is like so **Unfocused:** "Filter by event..." **Focused:** "Listening for input..."

**Shortcuts Search**

https://user-images.githubusercontent.com/41730826/193463380-a9d6acf9-6a39-4281-a5f8-a07c9b6d465e.mp4

**Input map Search**

https://user-images.githubusercontent.com/41730826/193463703-2b84f757-c9af-4320-9d9c-349004622d53.mp4

**New Event Picker (no more tabs!) (can now pick escape from the keyboard and close the window with escape too!)**

https://user-images.githubusercontent.com/41730826/193463415-af864b47-d634-4814-aaa1-4fa1bacb2d4c.mp4

### CODE REVIEW

File: editor/action_map_editor.cpp [hunk 1]
```diff
@@ -35,6 +35,100 @@
 #include "editor/editor_scale.h"
```
Comment (EricEzaM) [inline editor/action_map_editor.cpp:35]: Might be time to split this file up. It contains 3 classes now. I would do it in another pr tho.

**Review by bruvzg — COMMENTED** [review]
Seems like a good feature to have:

A minor issue - it's possible to input text input the filter `LineEdit` if you keep pressing the same letter key, e.g., pressing <kbd>F</kbd> multiple type will result in `fffffffF` in the filter.

Also, I'm not sure about "Filter by event... (perform input here)", it's a bit too long. Maybe it's better to display only `Filter by event...` by default and change it to `Listening for input...` when it's focused, or something like this.

**Comment by EricEzaM** [comment]
> Also, I'm not sure about "Filter by event... (perform input here)", it's a bit too long. Maybe it's better to display only `Filter by event...` by default and change it to `Listening for input...` when it's focused, or something like this.
 
What a great idea, can't believe I didn't think of this!

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by kristin-meshka** [comment]
Don't know if this is really the right place too put this, but i found a bug with the input search. That when you press Esc it will close the window while listening for inputs.

https://user-images.githubusercontent.com/12241212/208652480-014d41fe-8711-457f-924e-0df469330770.mp4

**Comment by KoBeWi** [comment]
> Don't know if this is really the right place too put this

It's not, open a new issue.
